src/Codeception/Lib/Generator/GherkinSnippets.php

```diff
@@ -16,22 +16,27 @@ public function {{methodName}}({{params}})
      {
         throw new \Codeception\Exception\Incomplete("Step `{{text}}` is not defined");
      }
-     
+
 EOF;
 
     protected $snippets = [];
     protected $processed = [];
 
-    public function __construct($settings)
+    public function __construct($settings, $test = null)
     {
         $loader = new Gherkin($settings);
+        if (empty($test)) {
+            $pattern = $loader->getPattern();
+        } else {
+            $pattern = $test;
+        }
 
         $finder = Finder::create()
             ->files()
             ->sortByName()
             ->in($settings['path'])
             ->followLinks()
-            ->name($loader->getPattern());
+            ->name($pattern);
 
         foreach ($finder as $file) {
             $pathname = str_replace("//", "/", $file->getPathname());
```

Comment:
I've created a new (local) branch https://github.com/edno/Codeception/tree/pr-3216 that contains the code refactored into 3 commits:
- New param test
- Extend param test for supporting folder
- Additional tests
@DavertMik if ok with you, I'll close this PR and create a new one into the branch 2.2